### ISSUE TITLE
Dispatch is suddenly not working anywhere for me

### ISSUE THREAD

**Issue by jsatk** [tpope/vim-dispatch#243]
Not sure why, but at some point in the past few months Dispatch has refused to work. If I run a simple `:Dispatch ls` in Terminal.app I get `(sync/?)`, in iTerm I get `(iterm/?)` and in tmux (where I normally spend my days) I get `(tmux/?)`. 

I've dug through the issues here and am at a loss on what could've suddenly caused Dispatch to seemingly refuse to do anything. 

I'm on the latest version of macOS, Vim 8, tmux, iTerm 2, etc. 

Any help in getting Dispatch working for me again would be much appreciated.

**Comment by tpope** [comment]
You can try `:let g:dispatch_fifo_callback = 0` to rule out the new FIFO usage. If it's still broken with that set, I'm afraid you're going to have to bisect.

**Comment by jsatk** [comment]
Setting `:let g:dispatch_fifo_callback = 0` causes the quickfix window to stay open when `:Dispatch ls` finishes. However I'm still seeing the `(tmux/?)` business. 

![screen shot 2018-08-20 at 1 43 49 pm](https://user-images.githubusercontent.com/1005550/44365599-26968280-a47f-11e8-9128-f80c4976fc87.png)

**Comment by tpope** [comment]
Curious but not particularly revealing. Time to bisect. Start with `v1.4`.

**Comment by jsatk** [comment]
@tpope I figured out what the issue is.

I use fish shell. I used to have this snippet in my vimrc.

```vim
if &shell =~# 'fish$'
  set shell=sh
endif
```

I removed it (see the commit [here](https://github.com/jsatk/dotfiles/commit/2d00ddc8d695b523be3a24a22088f16b2ad786ba?diff=unified#diff-076d61938d25fd036d6436c94d8778faL27) if you're curious) because I saw you had a similar line in vim-sensible [here](https://github.com/tpope/vim-sensible/blob/master/plugin/sensible.vim#L66-L68) — which I use.

For reasons I don't understand it appears that line from `vim-sensible` does not do the same thing my snippet did that I removed. It's possible it could be it's involved in vim-sensible later than I want it to.

Would love to figure this out so I can remove Fish shell check from my vimrc and leverage yours in vim-sensible. 

Thanks for your responses and help @tpope.
